home-application/src/main/java/com/softserveinc/ita/homeproject/application/api/CooperationApiImpl.java

```diff
@@ -0,0 +1,184 @@
+package com.softserveinc.ita.homeproject.application.api;
+
+import static com.softserveinc.ita.homeproject.application.constants.Permissions.CREATE_COOPERATION_PERMISSION;
+import static com.softserveinc.ita.homeproject.application.constants.Permissions.CREATE_HOUSE_PERMISSION;
+import static com.softserveinc.ita.homeproject.application.constants.Permissions.DEACTIVATE_COOPERATION_PERMISSION;
+import static com.softserveinc.ita.homeproject.application.constants.Permissions.DEACTIVATE_HOUSE_PERMISSION;
+import static com.softserveinc.ita.homeproject.application.constants.Permissions.GET_ALL_COOPERATION_PERMISSION;
+import static com.softserveinc.ita.homeproject.application.constants.Permissions.GET_COOPERATION_PERMISSION;
+import static com.softserveinc.ita.homeproject.application.constants.Permissions.GET_HOUSES_PERMISSION;
+import static com.softserveinc.ita.homeproject.application.constants.Permissions.GET_HOUSE_PERMISSION;
+import static com.softserveinc.ita.homeproject.application.constants.Permissions.UPDATE_COOPERATION_PERMISSION;
+import static com.softserveinc.ita.homeproject.application.constants.Permissions.UPDATE_HOUSE_PERMISSION;
+
+import java.math.BigDecimal;
+import java.util.HashMap;
+import java.util.Map;
+import javax.validation.Valid;
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.Provider;
+
+import com.softserveinc.ita.homeproject.api.CooperationApi;
+import com.softserveinc.ita.homeproject.homeservice.dto.CooperationDto;
+import com.softserveinc.ita.homeproject.homeservice.dto.HouseDto;
+import com.softserveinc.ita.homeproject.homeservice.service.CooperationService;
+import com.softserveinc.ita.homeproject.homeservice.service.HouseService;
+import com.softserveinc.ita.homeproject.model.CreateCooperation;
+import com.softserveinc.ita.homeproject.model.CreateHouse;
+import com.softserveinc.ita.homeproject.model.ReadCooperation;
+import com.softserveinc.ita.homeproject.model.ReadHouse;
+import com.softserveinc.ita.homeproject.model.UpdateCooperation;
+import com.softserveinc.ita.homeproject.model.UpdateHouse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Component;
+
+@Provider
+@Component
+public class CooperationApiImpl extends CommonApi implements CooperationApi {
+
+    @Autowired
+    private CooperationService cooperationService;
+
+    @Autowired
+    private HouseService houseService;
+
+    @PreAuthorize(CREATE_COOPERATION_PERMISSION)
+    @Override
+    public Response createCooperation(@Valid CreateCooperation createCooperation) {
+        CooperationDto createCoopDto = mapper.convert(createCooperation, CooperationDto.class);
+        CooperationDto readCoopDto = cooperationService.createCooperation(createCoopDto);
```

Comment:
It would be appropriate to return ReadCooperation object in the createCooperation method, so we will remove unnecessary mappings and remove 2 lines

Reply:
We will mix layers if we'll remove ReadCooperation object to service layers. Presentation layer and service layer must not know about each other